src/omnicore_rpctx.cpp

```diff
@@ -0,0 +1,873 @@
+            "PropertyIDForSale   : the property to list for sale (must be MSC or TMSC)\n"
+            "AmountForSale       : the amount to list for sale\n"
+            "AmountDesired       : the amount of BTC desired\n"
+            "PaymentWindow       : the time limit a buyer has to pay following a successful accept\n"
+            "MinAcceptFee        : the mining fee a buyer has to pay to accept\n"
+            "Action              : the action to take: (1) new, (2) update, (3) cancel \n"
+            "Result:\n"
+            "txid    (string) The transaction ID of the sent transaction\n"
+            "\nExamples:\n"
+            ">omnicored senddexsell_OMNI \"1FromAddress\" PropertyIDForSale \"AmountForSale\" \"AmountDesired\" PaymentWindow MinAcceptFee Action\n"
+        );
+
+    // obtain parameters & info
+    std::string fromAddress = params[0].get_str();
+    unsigned int propertyIdForSale = int64Touint32Safe(params[1].get_int64());
+    string strAmountForSale = params[2].get_str();
+    string strAmountDesired = params[3].get_str();
+    int64_t paymentWindow = params[4].get_int64();
+    int64_t minAcceptFee = StrToInt64(params[5].get_str(), true); // BTC so always divisible
+    int64_t action = params[6].get_int64();
+    const int64_t senderBalance = getMPbalance(fromAddress, propertyIdForSale, BALANCE);
+    const int64_t senderAvailableBalance = getUserAvailableMPbalance(fromAddress, propertyIdForSale);
+
+    // perform conversions
+    int64_t amountForSale = 0, amountDesired = 0;
+    if ((propertyIdForSale > 2 || propertyIdForSale <=0)) throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid propertyID for sale - only 1 and 2 are permitted");
+    amountForSale = StrToInt64(strAmountForSale, true); // TMSC/MSC always divisible
+    amountDesired = StrToInt64(strAmountDesired, true); // BTC so always divisible
+
+    // perform checks
+    if (action <= 0 || 3 < action) throw JSONRPCError(RPC_TYPE_ERROR, "Invalid action (1,2,3 only)");
+    if (action <= 2) { // actions 3 permit zero values, skip check
+        if (0 >= amountForSale) throw JSONRPCError(RPC_TYPE_ERROR, "Invalid amount for sale");
+        if (!isRangeOK(amountForSale)) throw JSONRPCError(RPC_TYPE_ERROR, "Amount for sale not in range");
+        if (0 >= amountDesired) throw JSONRPCError(RPC_TYPE_ERROR, "Invalid amount desired");
+        if (!isRangeOK(amountDesired)) throw JSONRPCError(RPC_TYPE_ERROR, "Amount desired not in range");
+    }
+    if (action != 3) { // only check for sufficient balance for new/update sell offers
+        if (senderBalance < amountForSale) throw JSONRPCError(RPC_TYPE_ERROR, "Sender has insufficient balance");
+        if (senderAvailableBalance < amountForSale) throw JSONRPCError(RPC_TYPE_ERROR, "Sender has insufficient balance (due to pending transactions)");
+    }
+    if (minAcceptFee <= 0) throw JSONRPCError(RPC_TYPE_ERROR, "Mininmum accept mining fee invalid");
```

Comment:
> What's wrong about no fee?

It's illegal per spec?  The field is 'number of coins' which requires a positive value.